pkg/controllermanager/controller/shoot/state/finalizer/reconciler_test.go

```diff
@@ -0,0 +1,277 @@
+// SPDX-FileCopyrightText: SAP SE or an SAP affiliate company and Gardener contributors
```

Comment:
We usually don't write unit tests for the `Reconcile` logic of controllers but only envtests/integrations

Reply:
Thanks for the shared insights, you can find the integration tests updated in my latest changes.